component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -444,6 +500,26 @@ private void validateAccountLockStatusForLocalUser(AuthenticationContext context
             if (log.isDebugEnabled()) {
                 log.debug(errorMessage);
             }
+            String accountLockedReason = StringUtils.EMPTY;
+            try {
+                UserRealm userRealm = TOTPUtil.getUserRealm(username);
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+                Map<String, String> claimValues = userStoreManager.getUserClaimValues(
+                        IdentityUtil.addDomainToName(authenticatedUserObject.getUserName(),
+                                authenticatedUserObject.getUserStoreDomain()),
+                        new String[]{ TOTPAuthenticatorConstants.ACCOUNT_LOCKED_REASON_CLAIM_URI},
+                        UserCoreConstants.DEFAULT_PROFILE);
+                if (claimValues != null) {
+                    accountLockedReason = claimValues.get(TOTPAuthenticatorConstants.ACCOUNT_LOCKED_REASON_CLAIM_URI);
+                }
+            }
+            catch (UserStoreException e) {
+                throw new AuthenticationFailedException(errorMessage + " Could not get the account locked reason.");
+            }
+            IdentityErrorMsgContext customErrorMessageContext = new IdentityErrorMsgContext(
+                    UserCoreConstants.ErrorCode.USER_IS_LOCKED +
+                    ":" + accountLockedReason);
```

Comment:
Can concat these two lines to one

Reply:
Addressed in https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/156